### PR TITLE
feat: add webm support

### PR DESCRIPTION
**Pull request by oupala**
Here is a first try to add webm support to downsize.

I'm not sure where I had to add the new option.

Not sure either what test could be added.

Feel free to discuss and tell me what could be improved.

### CODE REVIEW

**Comment by rprieto** [comment]
Thanks for the PR, it looks great! The code is perfect as-is. Could you just add:

- an example in the README like `{ format: "webp" }` to make it very explicit?
- for the tests, I currently have integration tests that process a source file and check that the output is as expected. Can you try looking at one of the mp4 examples and see if we can make a similar one for webp?

**Comment by oupala** [comment]
- I have added an example in the README.
- I have also added a test but it fails.
  - Could you please help me understand why it fails so that I can fix it?

And finally, note that the export format is `webm` not not `webp`. `webm` is for video while `webp` is for picture.

**Comment by rprieto** [comment]
Ha yes good catch thanks! The README and test look good :+1: What's the output of `npm test`?

**Comment by oupala** [comment]
There is something linked to an image, while I try to convert a video. I think that your test tries to extract an image from the converted video. But I don't understrand how it is done.

> can convert a video to webm
> metadata
> File not found: test-data/expected/videos/short.webm.jpg
>     ✖ exiftool failed to get metadata

I'm converting `short.mp4` to `short.webm`. What is this file `short.webm.jpg` and where does it come from?

**Comment by oupala** [comment]
I need help on how to perform test for video convertion from mp4 to webm.

Why is the test about an image (jpg) instead of a video (webm)?

As I don't understand how it should work, it's difficult for me to make a clean pull request.

**Comment by rprieto** [comment]
Hi, sorry for the delay some other things came up.

In order to check that the video was converted successfully, it takes a screenshot and compares it to a screenshot of the original. Now that you point it out, I think the original screenshot is assumed to be there (instead of being re-computed on the fly).

The test should have generated a screenshot of your `webm` video in the `actual` folder. Can you please copy it to the `expexted` folder so it acts as the reference?

In the future the tests could be better but for now this should do!

**Comment by oupala** [comment]
Ok, everything is green now. I think the merge request can be merged now.

The only strange thing is - in my opinion - to have a test such as `expect: 'videos/short.webm',` but to have no file named `videos/short.webm` in `test-data` (only `videos/short.webm.jpg`).

**Comment by rprieto** [comment]
Hi, thanks a lot for your patience I'll release a new version as soon as I can. The next step will be to add a flag to Thumbsup like `--video-format`.

I agree about the tests, it doesn't feel quite right. Will hopefully find time to clean that up a bit.

File: test/video.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ tape('can downsample a video for a smaller filesize', (test) => {
   })
 })
 
+tape('can convert a video to webm', (test) => {
+  diff.video(test, {
+    input: 'videos/short.mp4',
+    expect: 'videos/short.webm',
+    options: {}
```
Comment (rprieto) [inline test/video.js]: I just noticed, should this actually be

```
options: {
  format: 'webm'
}
```

?

Comment (oupala) [reply test/video.js]: I added the option.

I wasn't aware of that...

Comment (oupala) [reply test/video.js]: The bad thing is that the test was successful without the option...

That means that the test is not testing correctly...

Comment (rprieto) [reply test/video.js]: Good point. We should probably have a way to assert on some of the expected video metadata.

**Pull request merged**